README.md
# [toasty](https://pub.dartlang.org/packages/toasty)
[![pub package](version_image.svg)](https://pub.dartlang.org/packages/toasty)

A Flutter Library For Show Toast In Android And Ios
> Supported Platforms
>
> - Android
> - IOS

## How Use Toasty 

```yaml
# add the below line to your pubspec.yaml dependencies
toasty: 0.0.3
```

```dart
// add the below line in your dart code
import 'package:toasty/toasty.dart';
```
```dart
Toasty.showToast(
    message: 'this is a message',
    length: TOAST_LENGTH.LENGTH_SHORT,
    gravity: TOAST_GRAVITY.BOTTOM,
    fontColor: Colors.white,
    backgroundColor: Colors.grey.shade300,
    fontSize: 15,
  );
```
<br/>

### Toasty.showToast() property :
| property        | description                                                        |
| --------------- | ------------------------------------------------------------------ |
| message             | String (required)                                        |
| length     | TOAST_LENGTH.LENGTH_SHORT or TOAST_LENGTH.LENGTH_LONG <br/>(optional but defualt is TOAST_LENGTH.LENGTH_SHORT)                 |
| gravity         | TOAST_GRAVITY.BOTTOM or TOAST_GRAVITY.CENTER or TOAST_GRAVITY.TOP  <br/>(optional but defualt is TOAST_GRAVITY.BOTTOM ) |
| fontColor | Colors.white <br/> (if passing null to fontColor if your device is iOs defualt fontColor is White but if your device is Android your defualt fontColor is Black                                                 |
| backgroundColor         | Colors.grey  <br/> (if passing null to backgroundColor if your device is iOs defualt backgroundColor is Colors.black.withOpacity(0.9) but if your device is Android your defualt backgroundColor is Colors.grey.shade300                                                     |
| fontSize       | 16.0 (double) <br/> (defualt value is 15.0)                                                      |
| timeInSecForIos       | for length of toast in iOS (second) (defualt value is 1)                                              |

<br/>

##### if you want to use show success message you can use Toasty.successToast() this function show a green toast with a success icon in left of the text and for error and warning also you can use Toasty.errorToast() and Toasty.warningToast().

### Toasty.successToast() property :
| property        | description                                                        |
| --------------- | ------------------------------------------------------------------ |
| message             | String (required)                                        |
| length     | TOAST_LENGTH.LENGTH_SHORT or TOAST_LENGTH.LENGTH_LONG <br/>(optional but defualt is TOAST_LENGTH.LENGTH_SHORT)                 |
| gravity         | TOAST_GRAVITY.BOTTOM or TOAST_GRAVITY.CENTER or TOAST_GRAVITY.TOP  <br/>(optional but defualt is TOAST_GRAVITY.BOTTOM ) |
| fontColor | Colors.white <br/> (if passing null to fontColor if your device is iOs defualt fontColor is White but if your device is Android your defualt fontColor is Black                                                 |
| fontSize       | 16.0 (double) <br/> (defualt value is 15.0)                                                      |
| iconSize       | the size of the icon in toast (double) <br/> (defualt value is 16.0)                                                      |
| iconColor       | color of the icon in toast <br/> (defualt value is Colors.white)                                                |

this properties also use for Toasty.errorToast() and Toasty.warningToast()
 
<br/>

### To cancel all the toasts call

```dart
Toasty.cancel()
```
<br/>

## Preview Images

<img src="https://raw.githubusercontent.com/AliHoseinpoor/toasty/master/screenshot/1.jpg" width="300px" />                                                                                                                
<img src="https://raw.githubusercontent.com/AliHoseinpoor/toasty/master/screenshot/2.jpg" width="300px" />                                                                                                                
<img src="https://raw.githubusercontent.com/AliHoseinpoor/toasty/master/screenshot/3.jpg" width="300px" />                                                                                                                
<img src="https://raw.githubusercontent.com/AliHoseinpoor/toasty/master/screenshot/4.jpg" width="300px" />                                                                                                                
<img src="https://raw.githubusercontent.com/AliHoseinpoor/toasty/master/screenshot/5.jpg" width="300px" />                                                                                                                
<img src="https://raw.githubusercontent.com/AliHoseinpoor/toasty/master/screenshot/6.jpg" width="300px" />                                                                                                                
<img src="https://raw.githubusercontent.com/AliHoseinpoor/toasty/master/screenshot/7.jpg" width="300px" /> 

<br/>

## Functions

```dart
Future<void> cancel();
```

```dart
Future<void> showToast();
```

```dart
Future<void> successToast();
```

```dart
Future<void> errorToast();
```

```dart
Future<void> warningToast();
```


                                                                                                               



CHANGELOG.md
## toasty 0.0.3
bug fix in java code and improve performance

## toasty 0.0.2
some bug fix in this version
## toasty 0.0.1

## Platform Specifics
#### The following functions available on Android:<br/>
Toasty.cancel()<br/>
Toasty.showToast()<br/>
Toasty.successToast()<br/>
Toasty.warningToast()<br/>
Toasty.errorToast()<br/>

#### and The following functions available on iOS:<br/>
Toasty.cancel()<br/>
Toasty.showToast()<br/>


test/toasty_test.dart
import 'package:flutter/services.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:toasty/toasty.dart';

void main() {
  const MethodChannel channel = MethodChannel('toasty');

  setUp(() {
    channel.setMockMethodCallHandler((MethodCall methodCall) async {
      return '42';
    });
  });

  tearDown(() {
    channel.setMockMethodCallHandler(null);
  });

  test('getPlatformVersion', () async {
  });
}


example/test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:toasty_example/main.dart';

void main() {
  testWidgets('Verify Platform version', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(MyApp());

    // Verify that platform version is retrieved.
    expect(
      find.byWidgetPredicate(
        (Widget widget) => widget is Text &&
                           widget.data.startsWith('Running on:'),
      ),
      findsOneWidget,
    );
  });
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:toasty/toasty.dart';

void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Toasty'),
        ),
        body: Center(
          child: new Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              new RaisedButton(
                onPressed: () async {
                  Toasty.cancel();
                },
                child: new Text('Cancel Toast'),
              ),
              new SizedBox(
                height: 5,
              ),
              new RaisedButton(
                onPressed: () async {
                  Toasty.showToast(
                    message: 'this is a message',
                    length: TOAST_LENGTH.LENGTH_SHORT,
                    gravity: TOAST_GRAVITY.BOTTOM,
                    fontSize: 15,
                  );
                },
                child: new Text('Bottom Short Toast'),
              ),
              new SizedBox(
                height: 5,
              ),
              new RaisedButton(
                onPressed: () async {
                  Toasty.showToast(
                    message: 'this is a message',
                    length: TOAST_LENGTH.LENGTH_LONG,
                    gravity: TOAST_GRAVITY.BOTTOM,
                    fontSize: 15,
                  );
                },
                child: new Text('Bottom Long Toast'),
              ),
              new SizedBox(
                height: 5,
              ),
              new RaisedButton(
                onPressed: () async {
                  Toasty.showToast(
                    message: 'this is a message',
                    fontSize: 15,
                    gravity: TOAST_GRAVITY.CENTER,
                  );
                },
                child: new Text('Center Toast'),
              ),
              new SizedBox(
                height: 5,
              ),
              new RaisedButton(
                onPressed: () async {
                  Toasty.showToast(
                    message: 'this is a message',
                    fontSize: 15,
                    gravity: TOAST_GRAVITY.TOP,
                  );
                },
                child: new Text('Top Toast'),
              ),
              new SizedBox(
                height: 5,
              ),
              new RaisedButton(
                onPressed: () async {
                  Toasty.showToast(
                    message: 'this is a message',
                    length: TOAST_LENGTH.LENGTH_SHORT,
                    gravity: TOAST_GRAVITY.BOTTOM,
                    fontColor: Colors.white,
                    backgroundColor: Colors.black,
                    fontSize: 18,
                  );
                },
                child: new Text('Custom Toast'),
              ),
              new SizedBox(
                height: 5,
              ),
              new RaisedButton(
                onPressed: () async {
                  Toasty.successToast(
                    message: 'this is a success message',
                    length: TOAST_LENGTH.LENGTH_SHORT,
                    gravity: TOAST_GRAVITY.BOTTOM,
                    fontSize: 17,
                    iconSize: 17,
                    iconColor: Colors.white,
                    fontColor: Colors.white,
                  );
                },
                child: new Text('Success Toast'),
              ),
              new SizedBox(
                height: 5,
              ),
              new RaisedButton(
                onPressed: () async {
                  Toasty.errorToast(
                    message: 'this is a error message',
                    length: TOAST_LENGTH.LENGTH_SHORT,
                    gravity: TOAST_GRAVITY.BOTTOM,
                    fontSize: 17,
                    iconSize: 17,
                    iconColor: Colors.white,
                    fontColor: Colors.white,
                  );
                },
                child: new Text('Error Toast'),
              ),
              new SizedBox(
                height: 5,
              ),
              new RaisedButton(
                onPressed: () async {
                  Toasty.warningToast(
                    message: 'this is a warning message',
                    length: TOAST_LENGTH.LENGTH_SHORT,
                    gravity: TOAST_GRAVITY.BOTTOM,
                    fontSize: 17,
                    iconSize: 17,
                    iconColor: Colors.white,
                    fontColor: Colors.white,
                  );
                },
                child: new Text('Warning Toast'),
              ),
              new SizedBox(
                height: 5,
              ),
            ],
          ),
        ),
      ),
    );
  }
}


ios/Classes/ToastyPlugin.m
#import "ToastyPlugin.h"
#import "UIView+Toast.h"

static NSString *const CHANNEL_NAME = @"toasty";

@interface ToastyPlugin ()
@property(nonatomic, retain) FlutterMethodChannel *channel;
@end

@implementation ToastyPlugin{
  FlutterResult _result;
}

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:CHANNEL_NAME
            binaryMessenger:[registrar messenger]];
  UIViewController *viewController =
        [UIApplication sharedApplication].delegate.window.rootViewController;
  ToastyPlugin* instance = [[ToastyPlugin alloc] init];
  instance.channel = channel;
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (UIColor*) colorWithHex: (NSUInteger)hex {
    CGFloat red, green, blue, alpha;

    red = ((CGFloat)((hex >> 16) & 0xFF)) / ((CGFloat)0xFF);
    green = ((CGFloat)((hex >> 8) & 0xFF)) / ((CGFloat)0xFF);
    blue = ((CGFloat)((hex >> 0) & 0xFF)) / ((CGFloat)0xFF);
    alpha = hex > 0xFFFFFF ? ((CGFloat)((hex >> 24) & 0xFF)) / ((CGFloat)0xFF) : 1;

    return [UIColor colorWithRed: red green:green blue:blue alpha:alpha];
}

- (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
    if([@"cancel" isEqualToString:call.method]) {
        [[UIApplication sharedApplication].delegate.window.rootViewController.view hideAllToasts];
        result(void);
    } else if ([@"showToast" isEqualToString:call.method]) {
        NSString *msg = call.arguments[@"message"];
        NSString *gravity = call.arguments[@"gravity"];
        NSString *durationTime = call.arguments[@"time"];
        NSNumber *bgcolor = call.arguments[@"back_color"];
        NSNumber *textcolor = call.arguments[@"font_color"];
        NSNumber *fontSize = call.arguments[@"font_size"];

        CGFloat cgf = [fontSize doubleValue];
        int time = 1;
        @try {
            time = [durationTime intValue];
        } @catch (NSException *e) {
            time = 3;
        }

        if (time > 10) time = 10;
        else if (time < 1) time = 1;


        CSToastStyle *style = [[CSToastStyle alloc] initWithDefaultStyle];
        style.messageFont = [UIFont systemFontOfSize:cgf];
        style.backgroundColor = [self colorWithHex:bgcolor.unsignedIntegerValue];
        style.messageColor = [self colorWithHex:textcolor.unsignedIntegerValue];

        if (@available(iOS 11.0, *)) {
            UIWindow *window = UIApplication.sharedApplication.keyWindow;
            CGFloat topPadding = window.safeAreaInsets.top;
            CGFloat bottomPadding = window.safeAreaInsets.bottom;
        }

        if ([gravity isEqualToString:@"top"]) {
            [[[UIApplication sharedApplication].windows lastObject] makeToast:msg
                                                                                    duration:time
                                                                                    position:CSToastPositionTop
                                                                                       style:style
                                                                                       ];
        } else if ([gravity isEqualToString:@"center"]) {
            [[[UIApplication sharedApplication].windows lastObject] makeToast:msg
                                                                                        duration:time
                                                                                        position:CSToastPositionCenter
                                                                                           style:style
                                                                                          ];
        } else {
            [[[UIApplication sharedApplication].windows lastObject] makeToast:msg
                                                                                        duration:time
                                                                                        position:CSToastPositionBottom
                                                                                           style:style
                                                                                           ];
        }
        result(void);

    } else {
        result(FlutterMethodNotImplemented);
    }
}

@end


toasty/android/src/main/java/alihoseinpoor/com/toasty/ToastyPlugin.java
package alihoseinpoor.com.toasty;

import android.annotation.SuppressLint;
import android.content.Context;
import android.graphics.PorterDuff;
import android.graphics.drawable.Drawable;
import android.os.Build;
import android.util.TypedValue;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;
import java.util.Objects;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/**
 * ToastyPlugin
 */
public class ToastyPlugin  implements MethodCallHandler {
    private Context context;
    private Toast toast;
    private View view;

    @SuppressLint("InflateParams")
    private ToastyPlugin(Context context) {
        this.context = context;
        toast = new Toast(context);
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        if (inflater != null) {
            this.view = inflater.inflate(R.layout.toast_layout, null);
        }
    }

    public static void registerWith(Registrar registrar) {
        final MethodChannel channel = new MethodChannel(registrar.messenger(), "toasty");
        channel.setMethodCallHandler(new ToastyPlugin(registrar.context()));
    }

    private static void setIcon(View view, int id) {
        if (view != null) {
            ImageView imageView = (ImageView) view.findViewById(R.id.toastIcon);
            imageView.setImageResource(id);
        }
    }

    private static void setIconColor(View view, int color) {
        if (view != null) {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.FROYO) {
                ImageView imageView = (ImageView) view.findViewById(R.id.toastIcon);
                imageView.setColorFilter(color);
            }
        }
    }

    private static void setIconSize(View view, Number iconSize, Context context) {
        if (view != null) {
            if (iconSize != null) {
                int dpIconSize = Math.round(TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, iconSize.intValue(), context.getResources().getDisplayMetrics()));
                ImageView imageView = (ImageView) view.findViewById(R.id.toastIcon);
                imageView.getLayoutParams().height = dpIconSize;
                imageView.getLayoutParams().width = dpIconSize;
            }
        }
    }

    private static void setToastLength(Toast toasty, String length) {
        toasty.setDuration(length.equals("short") ? Toast.LENGTH_SHORT : Toast.LENGTH_LONG);
    }

    private static void setToastGravity(Toast toast, String gravity) {
        switch (gravity) {
            case "top":
                toast.setGravity(Gravity.TOP, 0, 100);
                break;
            case "center":
                toast.setGravity(Gravity.CENTER, 0, 0);
                break;
            default:
                toast.setGravity(Gravity.BOTTOM, 0, 100);
        }
    }

    private static View getToastView(Toast toast) {
        return (toast == null) ? null : toast.getView();

    }

    private static void setToastView(View view, Toast toast) {
        toast.setView(view);
    }

    private static void setToastText(Toast toast, String text, View view, int id) {
        if (toast != null && view != null) {
            TextView toastMessage = (TextView) view.findViewById(id);
            toastMessage.setText(text);
        }
    }

    private static void setToastMessageSize(View view, float size, int id) {
        if (view != null) {
            TextView toastMessage = (TextView) view.findViewById(id);
            toastMessage.setTextSize(size);
        }
    }

    private static void setToastMessageColor(View view, int color, int id) {
        if (view != null) {
            TextView toastMessage = (TextView) view.findViewById(id);
            toastMessage.setTextColor(color);
        }
    }

    private static void setToastDrawable(View view, Number color, Context context, boolean customToast, int id) {
        Drawable shapeDrawable;
        shapeDrawable = context.getResources().getDrawable(id);
        if (customToast) {
            if (color != null && shapeDrawable != null) {
                shapeDrawable.setColorFilter(color.intValue(), PorterDuff.Mode.SRC_IN);
            }
        }
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
            view.setBackground(shapeDrawable);
        }
    }

    private static void createTypedToast(MethodCall call, View view, Toast toast, Context context, int drawabl, int icon) {
        String message = Objects.requireNonNull(call.argument("message")).toString();
        String length = Objects.requireNonNull(call.argument("length")).toString();
        String gravity = Objects.requireNonNull(call.argument("gravity")).toString();
        Number fontSize = call.argument("font_size");
        Number fontColor = call.argument("font_color");
        Number iconColor = call.argument("icon_color");
        Number iconSize = call.argument("icon_size");

        ToastyPlugin.setToastView(view, toast);
        ToastyPlugin.setToastLength(toast, length);
        ToastyPlugin.setToastGravity(toast, gravity);

        if (view != null) {
            ToastyPlugin.setToastText(toast, message, view, R.id.toastMessage);
            if (fontSize != null) {
                ToastyPlugin.setToastMessageSize(view, fontSize.floatValue(), R.id.toastMessage);
            }
            if (fontColor != null) {
                ToastyPlugin.setToastMessageColor(view, fontColor.intValue(), R.id.toastMessage);
            }
            ToastyPlugin.setIcon(view, icon);
            if (iconColor != null) {
                ToastyPlugin.setIconColor(view, iconColor.intValue());
            }
            ToastyPlugin.setIconSize(view, iconSize, context);
            ToastyPlugin.setToastDrawable(view, null, context, false, drawabl);
        }

        toast.show();
    }


    @Override
    public void onMethodCall(MethodCall call, Result result) {
        switch (call.method) {
            case "showToast": {
                String message = Objects.requireNonNull(call.argument("message")).toString();
                String length = Objects.requireNonNull(call.argument("length")).toString();
                String gravity = Objects.requireNonNull(call.argument("gravity")).toString();
                Number fontSize = call.argument("font_size");
                Number fontColor = call.argument("font_color");
                Number backColor = call.argument("back_color");

                toast = Toast.makeText(context, message, length.equals("short") ? Toast.LENGTH_SHORT : Toast.LENGTH_LONG);
                View toastView = ToastyPlugin.getToastView(toast);

                if (toastView != null) {
                    ToastyPlugin.setToastText(toast, message, toastView, android.R.id.message);
                    ToastyPlugin.setToastView(toastView, toast);
                    ToastyPlugin.setToastGravity(toast, gravity);
                    ToastyPlugin.setToastDrawable(toastView, backColor, context, true, R.drawable.toast_drawable);
                    if (fontSize != null) {
                        ToastyPlugin.setToastMessageSize(toastView, fontSize.floatValue(), android.R.id.message);
                    }
                    if (fontColor != null) {
                        ToastyPlugin.setToastMessageColor(toastView, fontColor.intValue(), android.R.id.message);
                    }
                }
                toast.show();
                break;
            }

            case "cancel":
                if (toast != null) {
                    toast.cancel();
                }
                break;

            case "successToast":
                ToastyPlugin.createTypedToast(call, view, toast, context, R.drawable.toast_drawable_success, R.drawable.success);
                break;

            case "errorToast":
                ToastyPlugin.createTypedToast(call, view, toast, context, R.drawable.toast_drawable_error, R.drawable.error);
                break;

            case "warningToast":
                ToastyPlugin.createTypedToast(call, view, toast, context, R.drawable.toast_drawable_warning, R.drawable.warning);
                break;
        }
    }
}


lib/toasty.dart
import 'dart:async';
import 'dart:ui';
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

enum TOAST_LENGTH { LENGTH_SHORT, LENGTH_LONG }
enum TOAST_GRAVITY { BOTTOM, CENTER, TOP }

class Toasty {
  static const MethodChannel _channel = const MethodChannel('toasty');
  static String toastLength = 'short';
  static String toastGravity = 'bottom';

  static Future<void> cancel() async {
    await _channel.invokeMethod('cancel');
  }

  static Future<void> showToast({
    @required String message,
    TOAST_LENGTH length = TOAST_LENGTH.LENGTH_SHORT,
    TOAST_GRAVITY gravity = TOAST_GRAVITY.BOTTOM,
    double fontSize = 15.0,
    Color fontColor,
    Color backgroundColor,
    int timeInSecForIos = 1,
  }) async {
    toastLength = calculateLength(length);
    toastGravity = calculateGravity(gravity);

    if (fontColor == null &&
        backgroundColor == null &&
        defaultTargetPlatform == TargetPlatform.android) {
      fontColor = Colors.black;
      backgroundColor = Colors.grey.shade300;
    } else if (fontColor == null &&
        backgroundColor == null &&
        defaultTargetPlatform == TargetPlatform.iOS) {
      fontColor = Colors.white;
      backgroundColor = Colors.black.withOpacity(0.9);
    } else if (fontColor == null && defaultTargetPlatform == TargetPlatform.android) {
      fontColor = Colors.black;
    } else if (fontColor == null && defaultTargetPlatform == TargetPlatform.iOS) {
      fontColor = Colors.white;
    } else if (backgroundColor == null && defaultTargetPlatform == TargetPlatform.android) {
      backgroundColor = Colors.grey.shade300;
    } else if (backgroundColor == null && defaultTargetPlatform == TargetPlatform.iOS) {
      backgroundColor = Colors.black.withOpacity(0.9);
    }

    final Map<String, dynamic> datas = <String, dynamic>{
      'message': message,
      'length': toastLength,
      'gravity': toastGravity,
      'font_size': fontSize,
      'font_color': fontColor.value,
      'back_color': backgroundColor.value,
      'time': timeInSecForIos,
    };

    await _channel.invokeMethod('showToast', datas);
  }

  static Future<void> successToast({
    @required String message,
    TOAST_LENGTH length = TOAST_LENGTH.LENGTH_SHORT,
    TOAST_GRAVITY gravity = TOAST_GRAVITY.BOTTOM,
    Color fontColor = Colors.white,
    double fontSize = 14.0,
    int iconSize = 16,
    Color iconColor = Colors.white,
  }) async {
    toastLength = calculateLength(length);
    toastGravity = calculateGravity(gravity);

    final Map<String, dynamic> datas = <String, dynamic>{
      'message': message,
      'length': toastLength,
      'gravity': toastGravity,
      'font_size': fontSize,
      'font_color': fontColor.value,
      'icon_color': iconColor.value,
      'icon_size': iconSize,
    };

    await _channel.invokeMethod('successToast', datas);
  }

  static Future<void> errorToast({
    @required String message,
    TOAST_LENGTH length = TOAST_LENGTH.LENGTH_SHORT,
    TOAST_GRAVITY gravity = TOAST_GRAVITY.BOTTOM,
    double fontSize = 14.0,
    int iconSize = 16,
    Color fontColor = Colors.white,
    Color iconColor = Colors.white,
  }) async {
    toastLength = calculateLength(length);
    toastGravity = calculateGravity(gravity);

    final Map<String, dynamic> datas = <String, dynamic>{
      'message': message,
      'length': toastLength,
      'gravity': toastGravity,
      'font_size': fontSize,
      'font_color': fontColor.value,
      'icon_color': iconColor.value,
      'icon_size': iconSize,
    };

    await _channel.invokeMethod('errorToast', datas);
  }

  static Future<void> warningToast({
    @required String message,
    TOAST_LENGTH length = TOAST_LENGTH.LENGTH_SHORT,
    TOAST_GRAVITY gravity = TOAST_GRAVITY.BOTTOM,
    double fontSize = 14.0,
    int iconSize = 16,
    Color fontColor = Colors.white,
    Color iconColor = Colors.white,
  }) async {
    toastLength = calculateLength(length);
    toastGravity = calculateGravity(gravity);

    final Map<String, dynamic> datas = <String, dynamic>{
      'message': message,
      'length': toastLength,
      'gravity': toastGravity,
      'font_size': fontSize,
      'font_color': fontColor.value,
      'icon_color': iconColor.value,
      'icon_size': iconSize,
    };

    await _channel.invokeMethod('warningToast', datas);
  }

  static String calculateLength(TOAST_LENGTH length) =>
      length == TOAST_LENGTH.LENGTH_LONG ? 'long' : 'short';

  static String calculateGravity(TOAST_GRAVITY gravity){
    if (gravity == TOAST_GRAVITY.CENTER) {
      return 'center';
    } else if (gravity == TOAST_GRAVITY.TOP) {
      return 'top';
    }else{
      return 'bottom';
    }
  }
}
